js/impress.js

```diff
@@ -639,10 +637,42 @@
         });
 
     };
-    
+
     // flag that can be used in JS to check if browser have passed the support test
-    impress.supported = impressSupported;
-    
+    Impress.supported = impressSupported;
+
+    // This is were the root elements of all impress.js instances will be kept.
+    // Yes, this means you can have more than one instance on a page, but I'm not
+    // sure if it makes any sense in practice ;)
+    Impress.roots = {};
+
+    // reset helper to reset one or all
+    Impress.reset = function( rootId ) {
+        var remove = function( rootId ) {
+            var root = byId( rootId );
+            root.parentNode.removeChild(root);
+            delete Impress.roots["impress-root-" + rootId];
+        };
+        if (rootId) {
+            remove( rootId );
+        } else {
+            Object.keys(Impress.roots).map(function(key) { return key.replace("impress-root-", "");}).forEach(remove);
+        }
+
+        body.className.split(' ').forEach(function(className) {
+            if (className.indexOf('impress-on-') === 0)
+                body.classList.remove(className);
+        });
+        location.hash = '';
+    };
+
+    // alias to support the backward compatible impress().init()
+    window.impress = function( rootId ) {
```

Comment:
That was definitely my mistake. I didn't realized that the code was still exposing the `impress` global here =(